--- Linux-PAM-0.72/modules/pam_limits/pam_limits.c~	Mon Mar 26 08:29:00 2001
+++ Linux-PAM-0.72/modules/pam_limits/pam_limits.c	Mon Mar 26 08:29:24 2001
@@ -474,9 +474,7 @@
             if (strcmp(uname, domain) == 0) /* this user have a limit */
                 process_limit(LIMITS_DEF_USER, ltype, item, value, ctrl);
             else if (domain[0]=='@') {
-		if (root)
-		    _pam_log(LOG_DEBUG, "group limits skipped for '%s'", uname);
-                else if (is_on_group(uname, domain+1))
+                if (!root && is_on_group(uname, domain+1))
                     process_limit(LIMITS_DEF_GROUP, ltype, item, value, ctrl);
             } else if (strcmp(domain, "*") == 0) {
 		if (root)
