diff -urN Linux-PAM-0.72/modules/pam_userdb.orig/Makefile Linux-PAM-0.72/modules/pam_userdb/Makefile
--- Linux-PAM-0.72/modules/pam_userdb.orig/Makefile	Sun Apr  1 01:42:32 2001
+++ Linux-PAM-0.72/modules/pam_userdb/Makefile	Sun Apr  1 01:42:11 2001
@@ -13,7 +13,7 @@
 LIBOBJD = $(addprefix dynamic/,$(LIBOBJ))
 #LIBOBJS = $(addprefix static/,$(LIBOBJ))
 
-EXTRALS += -ldb2
+EXTRALS += -ldb3
 
 dynamic/%.o : %.c
 	$(CC) $(CFLAGS) $(DYNAMIC) $(CPPFLAGS) $(TARGET_ARCH) -c $< -o $@
