Index: pam.deb/modules/pam_rhosts/pam_rhosts_auth.c
===================================================================
--- pam.deb.orig/modules/pam_rhosts/pam_rhosts_auth.c
+++ pam.deb/modules/pam_rhosts/pam_rhosts_auth.c
@@ -267,7 +267,7 @@
 
 
     /* Try for raw ip address first. */
-    if (isdigit(*lhost) && (int32_t)(laddr = inet_addr(lhost) != -1))
+    if (isdigit(*lhost) && (int32_t)(laddr = inet_addr(lhost)) != -1)
 	return (negate*(! (raddr ^ laddr)));
 
     /* Better be a hostname. */
